crypto/ocsp/ocsp_vfy.c

```diff
@@ -301,8 +304,11 @@ static int ocsp_match_issuerid(X509 *cert, OCSP_CERTID *cid,
             return -1;
         if (memcmp(md, cid->issuerNameHash.data, mdlen))
             return 0;
-        X509_pubkey_digest(cert, dgst, md, NULL);
-        if (memcmp(md, cid->issuerKeyHash.data, mdlen))
+        if (!X509_pubkey_digest(cert, dgst, md, NULL)) {
+            OCSPerr(0, OCSP_R_DIGEST_ERR);
+            return -1;
```

Comment:
Please at least document return values.

Reply:
I wish I knew what they meant :)

Reply:
I think in this function, it's -1 means error, 0 means no match, 1 means match.

Reply:
Comment added.

I also cleaned up the syntax and added additional errors on failure.